sycl/include/sycl/ext/oneapi/experimental/annotated_arg/annotated_arg.hpp

```diff
@@ -162,6 +162,46 @@ __SYCL_TYPE(annotated_arg) annotated_arg<T *, detail::properties_t<Props...>> {
   template <typename PropertyT> static constexpr auto get_property() {
     return property_list_t::template get_property<PropertyT>();
   }
+
+  template <typename T2, typename PropertyList>
```

Comment:
I initially thought this is the same propertyList as the one that templates this class. On closer look turns out that's not the case. Still rename it for clarity. We need these to work across different property lists, for example:
```c++
struct MyKernel {
  // NOTE: all arguments have different properties.
  annotated_arg<ac_int<...>, decltype(properties(A))> arg_a;
  annotated_arg<ac_int<...>, decltype(properties(B))> arg_b;
  annotated_arg<ac_int<...>, decltype(properties(C))> arg_c;
  void operator()() const {
    *arg_c = arg_a + arg_b;
  }
};
```
The properties for this class are only meaningful on the interface so it should be legal to allow arithmetic ops across objects with different property lists.

```suggestion
  template <typename T2, typename PropertyList2>
```

And similarly below and update the extension too.